pkgs/by-name/vm/vmctl/package.nix

```
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenvNoCC,
+  bash,
+  shellcheck,
+  fetchFromGitHub,
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "vmctl";
+  version = "v0.99-unstable-2024-05-14";
+
+  src = fetchFromGitHub {
+    owner = "SamsungDS";
+    repo = "vmctl";
+    rev = "5b6b7084b8cba06b474c0e020df965237f2c832c";
+    sha256 = "sha256-yDgaY2RJXBjWkMSQb4JaJzMGLFzowfOGixSRzzv2iIk=";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp vmctl $out/bin
+    cp -r {cmd,common,contrib,lib} $out
```

The program seems to have a few runtime dependencies that we should add to its `$PATH`. Upstream README seems to specify `ssh` and `socat`, but I found at least `tail`, `awk`, `cloud-localds` and `mkisofs` as well. I think `tail` (`coreutils`) might be given, but the others are probably not. Do you happen to know about any other runtime dependencies?

Remember to add the programs to args up top, as well as adding `makeWrapper` to `nativeBuildInputs` (and args)

```suggestion
    wrapProgram "$out/bin/vmctl" \
      --prefix PATH : "${lib.makeBinPath [ ssh socat gawk cloud-utils cdrtools ]}"

    cp -r {cmd,common,contrib,lib} $out
```

---

You are right. It also needs `qemu` and `qemu-tools`. I will also add that

---

@h7x4 I found this gist about wrapProgram [link](https://gist.github.com/CMCDragonkai/9b65cbb1989913555c203f4fa9c23374). Should I use --set instead of --prefix?

Plus, should I make part of postFixup or is it fine if it is a part of installPhase?

---

I used `set` instead `prefix`. It also helped find me all runtime dependencies. :)